src/EditorFeatures/Core.Wpf/SignatureHelp/Controller_NavigationKeys.cs

```diff
@@ -13,13 +13,29 @@ internal partial class Controller
         internal bool TryHandleUpKey()
         {
             AssertIsForeground();
-            return ChangeSelection(() => sessionOpt.PresenterSession.SelectPreviousItem());
+            var handled = ChangeSelection(() => sessionOpt.PresenterSession.SelectPreviousItem());
+
+            // If we're letting the editor handle it then we need to know an arrow key was pressed, as there is specific
+            // behaviour in that scenario
+            _caretMoveFromArrowKeys = !handled;
```

Comment:
oooh, i really don't like this.   gogni to have to see how this all fits together.  but generally very opposed to storing ephemeral state here in these types to be read out later.